src/storage/src/sink/kafka.rs

```diff
@@ -805,8 +805,16 @@ async fn determine_sink_resume_upper(
     .await
     .add_context("error registering kafka progress topic for sink")?;
 
+    // We are about to spawn a blocking task that cannot be aborted by simply calling .abort() on
+    // its handle but we must be able to cancel it prompty so as to not leave long running
+    // operations around when interest to this task is lost. To accomplish this we create a shared
+    // token of which a weak reference is given to the task and a strong reference is held by the
+    // parent task. The task periodically checks if its weak reference is still valid before
+    // continuing its work.
+    let parent_token = Arc::new(());
```

Comment:
note that if we ever need async checking for cancellation we may want to use https://docs.rs/tokio-util/latest/tokio_util/sync/struct.CancellationToken.html#

Reply:
very nice! Didn't know this existed